tests/test_downloadermiddleware_retry.py

```diff
@@ -104,5 +104,86 @@ def _test_retry_exception(self, req, exception):
         self.assertEqual(req, None)
 
 
+class MaxRetryTimesTest(unittest.TestCase):
+    def setUp(self):
+        self.crawler = get_crawler(Spider)
+        self.spider = self.crawler._create_spider('foo')
+        self.mw = RetryMiddleware.from_crawler(self.crawler)
+        self.mw.max_retry_times = 2
+
+    def test_without_metakey(self):
+        
+        req = Request('http://www.scrapytest.org/invalid_url')
+
+        # SETTINGS: RETRY_TIMES is NON-ZERO
+        self.mw.max_retry_times = 5
+        self._test_retry(req, DNSLookupError('foo'), 5)
+
+        # SETTINGS: RETRY_TIMES = 0
+        self.mw.max_retry_times = 0
+        self._test_retry(req, DNSLookupError('foo'), 0)
+
+        # RESET RETRY_TIMES SETTINGS
+        self.mw.max_retry_times = 2
+
+    def test_with_metakey_preceding(self):
+        # request with meta(max_retry_times) is called first
+        
+        req1 = Request('http://www.scrapytest.org/invalid_url', meta={'max_retry_times': 3})
+        req2 = Request('http://www.scrapytest.org/invalid_url')
+        req3 = Request('http://www.scrapytest.org/invalid_url', meta={'max_retry_times': 4})
+
+        # SETINGS: RETRY_TIMES < meta(max_retry_times)
+        self.mw.max_retry_times = 2
+        self._test_retry(req1, DNSLookupError('foo'), 3)
+        self._test_retry(req2, DNSLookupError('foo'), 2)
+
+        # SETINGS: RETRY_TIMES > meta(max_retry_times)
+        self.mw.max_retry_times = 5
+        self._test_retry(req3, DNSLookupError('foo'), 4)
+        self._test_retry(req2, DNSLookupError('foo'), 5)
+
+        # RESET RETRY_TIMES SETTINGS
+        self.mw.max_retry_times = 2
+
+    def test_with_metakey_succeeding(self):
+        # request with meta(max_retry_times) is called second
```

Comment:
Same as above, I do not understand what is meant by _"request with meta(max_retry_times) is called second"_